File: backend/src/app/api/v1/routes/rewind.py
from fastapi import APIRouter

router = APIRouter(prefix="/rewind", tags=["rewind"])


@router.get("/summary")
def rewind_summary():
    return {
        "user": {"name": "Demo User"},
        "topArtist": "Kendrick Lamar",
        "topSong": "HUMBLE.",
        "minutesPlayed": 12345,
    }

File: MusicRewind/app/rewind.ts
export type RewindSummary = {
  user: {
    name: string;
  };
  topArtist: string;
  topSong: string;
  minutesPlayed: number;
};

const API_BASE = process.env.EXPO_PUBLIC_API_URL;

export async function fetchRewindSummary(): Promise<RewindSummary> {
  if (!API_BASE) {
    throw new Error('API base URL not configured');
  }

  const res = await fetch(`${API_BASE}/api/v1/rewind/summary`);

  if (!res.ok) {
    throw new Error('Failed to load rewind data');
  }

  return res.json();
}